src/accelerate/utils/operations.py

```diff
@@ -287,9 +287,26 @@ def _gpu_gather_one(tensor):
         # Can only gather contiguous tensors
         if not tensor.is_contiguous():
             tensor = tensor.contiguous()
-        output_tensors = [torch.empty_like(tensor) for _ in range(torch.distributed.get_world_size())]
-        torch.distributed.all_gather(output_tensors, tensor)
-        return torch.cat(output_tensors, dim=0)
+
+        state = PartialState()
+
+        if state.backend is not None and state.backend != "gloo":
+            output_tensors = torch.zeros(
```

Comment:
Why use `torch.zeros` instead of `torch.empty_like` as previously?

Reply:
Added a comment